libc/src/signal/linux/signal.cpp

```diff
@@ -8,14 +8,16 @@
 
 #include "src/signal/signal.h"
 #include "hdr/signal_macros.h"
-#include "hdr/types/sighandler_t.h"
 #include "src/__support/common.h"
 #include "src/__support/macros/config.h"
 #include "src/signal/sigaction.h"
 
 namespace LIBC_NAMESPACE_DECL {
 
-LLVM_LIBC_FUNCTION(sighandler_t, signal, (int signum, sighandler_t handler)) {
+// Our LLVM_LIBC_FUNCTION macro doesn't handle function pointer return types.
+using signal_handler = void (*)(int);
```

Comment:
perhaps `decltype` would be better here?

Reply:
hmm...no nevermind. We'd only be able to do `decltype(signal)`, which would not be the type of the parameter or return value.

Reply:
Perhaps `auto` would work...